src/query/parser/promql/types.go

```diff
@@ -178,6 +239,10 @@ func promVectorCardinalityToM3(card promql.VectorMatchCardinality) logical.Vecto
 }
 
 func promMatchingToM3(vectorMatching *promql.VectorMatching) *logical.VectorMatching {
+	// vectorMatching can be nil iff at least one of the sides is a scalar
```

Comment:
nit: `s/iff/if`

Reply:
It's [intentional](https://en.wikipedia.org/wiki/If_and_only_if)